client/client-multi/client-c/src/kaa/kaa_notification_manager.h

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2014-2015 CyberVision, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef KAA_KAA_NOTIFICATION_H_
+#define KAA_KAA_NOTIFICATION_H_
+#include <stddef.h>
+#include <stdio.h>
+#include <stdint.h>
+#include "stdbool.h"
+#include "kaa_status.h"
+#include "kaa_error.h"
+#include "kaa_platform_common.h"
+#include "utilities/kaa_mem.h"
+#include "kaa_channel_manager.h"
+#include "kaa_common.h"
+#include "utilities/kaa_log.h"
+#include "kaa_platform_utils.h"
+#include "platform/ext_notification_receiver.h"
+
+
+/**
+* @brief Calculates the topic listener id to manage topic listeners.
+*
+* @param[in]   listener     Pointer to the listener, which id should be calculated.
+* @param[out]  listener_id  Pointer to the variable, which will be initialized with calculated id.
+*
+* @return Error code.
+*/
+kaa_error_t kaa_calculate_topic_listener_id(kaa_topic_listener_t *listener, uint32_t *listener_id);
+
+/**
+* @brief Calculates the notification listener id to manage topic listeners.
+*
+* @param[in]   listener     Pointer to the listener, which id should be calculated.
+* @param[out]  listener_id  Pointer to the variable, which will be initialized with calculated id.
+*
+* @return Error code.
+*/
+kaa_error_t kaa_calculate_notification_listener_id(kaa_notification_listener_t *listener, uint32_t *listener_id);
+
+/**
+* @brief Adds a mandatory notification listener.
```

Comment:
Adds a mandatory notification listener to receive notifications on mandatory topics.